src/lib/Libcmds/get_server.c

```diff
@@ -171,8 +171,8 @@ get_server(char *job_id_in, char *job_id_out, char *server_out)
 #else
 		strcat(job_id_out, parent_server);
 #endif /* localmod 086 */
-
-		strcpy(server_out, parent_server);
+		if (server_out[0] == '\0')
+			strcpy(server_out, parent_server);
```

Comment:
I am of the view that this whole if block can be skipped if we have copied current_server to server_out. The logic to append the server name to the job id is there below. What do you think?

Reply:
If we provide server2 in the pbs_conf of client on that time this logic will make sense as block of code above where it tries to match parent server and pbs_conf server, that block will not be executed and control will come here and if we skip this block the below block will do strcat between the job_id_out and the second server.